scripts/run_example

```diff
@@ -4,38 +4,76 @@ readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
 # shellcheck source=scripts/common
 source "$SCRIPTS_DIR/common"
 
-while getopts ":hC" opt; do
-    case "$opt" in
-    h)
-        echo -e "Usage: $0 [-c] [-h] [-C opts] example_name
-  -h    Print this message
-  -C    Pass subsequent options to example client"
-        exit 0
-        ;;
-    C)
-        echo "Passing remaining arguments to client"
-        break
-        ;;
-    \?)
-        echo "Invalid option -${OPTARG}"
-        exit 1
-        ;;
-    esac
-done
-shift $((OPTIND -1))
-
-readonly NAME="${*: -1}"
-if [[ -z "$NAME" ]]; then
-  echo "Missing example name, use -h for help"
+readonly NAME="${1}"
+if [[ -z "${NAME}" ]]; then
+  echo "Missing the example name, use -h for help"
   exit 1
 fi
+shift 1
+
+language="rust"
+server="asylo"
+while getopts "s:l:Ch" opt; do
+  case "${opt}" in
+  h)
+      echo -e "Usage: ${0} EXAMPLE_NAME [-h] [-l rust|cpp] [-s asylo|dev|asan|tsan|none] [-- CLIENT_ARGS]
+  -l    Example application version:
+          - rust (used by default)
+          - cpp
+  -s    Server type used to run examples:
+          - asylo: server based on the Asylo framework (used by default)
+          - dev: development version of the server
+          - asan: 'dev' server with address sanitizer
+          - tsan: 'dev' server with thread santizer
+          - none: run an application client without a server
+  -h    Print Help (this message) and exit"
+      exit 0;;
+  l)
+    language="${OPTARG}";;
+  s)
+    case "${OPTARG}" in
+    asylo|dev|asan|tsan|none)
+      server="${OPTARG}";;
+    *)
+      echo "Invalid server: ${OPTARG}"
+      exit 1;;
+    esac;;
+  *)
+    echo "Invalid argument: ${OPTARG}"
+    exit 1;;
+  esac
+done
+shift "$((OPTIND -1))"
 
-readonly CLIENT_ARGS=("${@:1:$((${#@}-1))}")
+if [[ "none" != "${server}" ]]; then
```

Comment:
(Personally, I'm not a fan of [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) but up to you.)